### ISSUE TITLE
Option to ignore slice indices in no-magic-numbers

### ISSUE THREAD

**Issue by l-abels** [eslint/eslint#13131]
**What rule do you want to change?**

`no-magic-numbers`

**Does this change cause the rule to produce more or fewer warnings?**

Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option: `ignoreSliceIndexes`

**Please provide some example code that this change will affect:**

```js
const rightTwo = arr.slice(-2);

const middle = arr.slice(1, -1);
```

**What does the rule currently do for this code?**

Slice indices are treated as magic numbers.

**What will the rule do after it's changed?**

Allow for arguments to `Array.slice` to be ignored as per array indices

**Are you willing to submit a pull request to implement this change?**

Yes, there are plenty of examples to work from.  I'm not sure how to determine whether the `slice` call belongs to an array or not.

**Comment by kaicataldo** [comment]
My concern with an option like this is that ESLint doesn't have access to type information, so there's no way to reliably ignore this case without false positives.
